scenes/controller/offline_controller/offline_controller.gd

```diff
@@ -0,0 +1,162 @@
+class_name OfflineController
+extends Node
+
+@export var worker_controller: WorkerController
+@export var enemy_controller: EnemyController
+
+###############
+## overrides ##
+###############
+
+
+func _ready() -> void:
+	_connect_signals()
+
+
+#############
+## helpers ##
+#############
+
+
+## workers must be "happy" to work while not observed (none of the resources are decreasing)
+## TODO: how to implement "unhappy" workers working while not observed? (oscillating resources, ...)
```

Comment:
Perhaps best course of action is to not calculate this case, because:
- requirements are high (adds complexity to offline controller)
- rewards are small (this scenario does not exist long term, having decreasing resources is discouraged)

Reply:
agreed